### PR TITLE
Cleanup/footstep parameters

### PR DESCRIPTION
**Pull request by PotatoPeeler3000**
Changes in this PR:
- Removed 10 parameters shown in image below
- Organized the setters, getters, and .json files so they all appear in the same order
- Renamed to DefaultFootStepPlannerParameters
- Added all the keys to the parameters file because it makes sense (its how we do it in other places)

### CODE REVIEW

File: ihmc-footstep-planning/src/main/java/us/ihmc/footstepPlanning/graphSearch/parameters/DefaultFootstepPlannerParameters.java [hunk 1]
```diff
@@ -1,17 +1,91 @@
 package us.ihmc.footstepPlanning.graphSearch.parameters;
 
+import us.ihmc.tools.property.BooleanStoredPropertyKey;
+import us.ihmc.tools.property.DoubleStoredPropertyKey;
+import us.ihmc.tools.property.IntegerStoredPropertyKey;
+import us.ihmc.tools.property.StoredPropertyKeyList;
 import us.ihmc.tools.property.StoredPropertySet;
 
-public class DefaultFootstepPlannerParameters extends StoredPropertySet implements FootstepPlannerParametersBasics
+public class DefaultFootstepPlannerParameters extends StoredPropertySet implements DefaultFootstepPlannerParametersBasics
```
Comment (PotatoPeeler3000) [inline ihmc-footstep-planning/src/main/java/us/ihmc/footstepPlanning/graphSearch/parameters/DefaultFootstepPlannerParameters.java:9]: I added the keys to this file, I found it confusing to put the parameters in a different file when this file is named DefaultFootStepPlannerParameters

**Review by PotatoPeeler3000 — COMMENTED** [review]
image

**Comment by PotatoPeeler3000** [comment]
Removed 10 parameters from the planner, tests still pass, when walking in sim I don't see any real difference when planning around and walking to goals. Did change the name to Default... because I was constantly getting confused between what was Default and not

**Review by stephenmcc — APPROVED** [review]

File: ihmc-footstep-planning/src/main/java/us/ihmc/footstepPlanning/graphSearch/stepChecking/FootstepPoseHeuristicChecker.java [hunk 2]
```diff
@@ -146,64 +143,24 @@ else if (Math.abs(stepHeight.getValue()) > maximumStepZ)
...
-      maxXFromPitchContraint.set(InterpolationTools.linearInterpolate(Math.abs(parameters.getMaximumStepReach()), parameters.getMaximumStepXWhenFullyPitched(), alphaPitchedBack));
-      double stepDownFraction = - stepHeight.getValue() / minZFromPitchContraint.getValue();
-      double stepForwardFraction = stepLength.getValue() / maxXFromPitchContraint.getValue();
+      double alphaPitchedBack = Math.max(0.0, - stanceFootPose.getPitch() / parameters.getMinSurfaceIncline());
 
       stepIsPitchedBack.set(alphaPitchedBack > 0.0);
-      stepTooLow.set(stepLength.getValue() > 0.0 && stepDownFraction > 1.0);
-      stepTooForward.set(stepHeight.getValue() < 0.0 && stepForwardFraction > 1.0);
+      stepTooLow.set(stepLength.getValue() > 0.0 && stepHeight.getValue() > 1.0);
+      stepTooForward.set(stepHeight.getValue() < 0.0 && stepLength.getValue() > 1.0);
```
Comment (rjgriffin42) [inline ihmc-footstep-planning/src/main/java/us/ihmc/footstepPlanning/graphSearch/stepChecking/FootstepPoseHeuristicChecker.java]: do these do anything anymore? As written, "stepTooLow" is now only true if you're stepping forward and up a meter in the air, which wasn't the intent of that. Also stepTooForward is a similar thing. So this should be either fixed, or these fields should be deleted and the `STEP_TOO_LOW_AND_FORWARD_WHEN_PITCHED` check should be removed.

**Pull request merged**